web-client/src/pages/routes/DashboardRoute.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import Location from 'react-app-location';
+import { Route, RouteProps } from 'react-router-dom';
+import { MenuItem } from 'src/types/menu-item';
+
+import DashboardLayout from '../../components/DashboardLayout/DashboardLayout';
+
+// TODO remove when using real profile state
+const mockProfileData = {
+  fullName: 'John Watkins',
+  photoUrl:
+    'https://lh3.googleusercontent.com/-6DkdLho7c5Y/AAAAAAAAAAI/AAAAAAAAAAA/AMZuuckYv30Z33APqrrucaWdEBbadI7WxQ/photo.jpg?sz=46',
+  followers: 15,
+  stars: 4.5,
+};
+
+// TODO change locations
+const siteLocations = {
+  logout: new Location('/requests/logout'),
+  contact: new Location('/requests/contact'),
+  map: new Location('/requests/map'),
+  notifications: new Location('requests/notifications'),
+};
```

Comment:
I've some ideas but it'll be better to follow some advice. As I see, the menuLinks and these siteLocations will be truly static, maybe we can find a way to make it only object like `siteLinks={ menuLinks: { profile: {logout: {location: '/logout'}}, request: {...}} general: {contact:{ location:'/contact'}. etc }}`.

Reply:
I like this idea. Having everything controlled by a JSON or a JS object, without needing to change the code to perform modifications or to rely on tricky automations

Reply:
I’d argue the code is easily understandable as is and the compiler will catch issues. I don’t see a need to abstract this away.

Reply:
yea, @btuerker this is fine, the only part I asked to be in configuration was actually the links and the menu items themselves.
You have done that correctly.
You don't have to generalize this part of the menu as it's part of the design and will remain the same.
although if you are generalizing this part as JSON, it should have been a format similar to the one used for the menu items themselves.

Reply:
However, there are a few issues with this again.
1. As per the designs, the center button should be for a new request and not for the map. You have currently put the link to the map there.
2. I think @telmodias wanted the new Request page to be modal and so it should change a state variable which should show the new request modal.
3. The notifications shouldn't be a page, it should be just like the side menu. The reason being, when people visit this website from a desktop browser, they would prefer it to be like a hovering menu that shows them the notifications rather than being redirected to a new page.

@puzzledbytheweb @telmodias could you please confirm these?

Reply:
@sabind what do you think about my opinion on the profile details, sign up buttons, etc being static? https://github.com/reach4help/reach4help/pull/481#discussion_r419085948
or do you think it should be a JSON, I think only the menu items should be a JSON

Reply:
Honestly maybe for now it’s best to keep explicit as it is already defined that way. I’m hesitant to keep increasing the file count for the sake of abstraction and it takes more effort to figure out where the code you need to change is buried (when you have a lot of code in between).

I like making it all explicit, but I also don’t want to impose. Just leaving my 2 cents.

Reply:
@sabind I am not sure what you meant by keeping it explicit.

In the case of the these items in the sidebar, it's static but the content is also pretty much fixed, but the designs themselves change based on the device and the type of user but not the modules.
So I think by converting these into JSON configs, we are trying to solve a problem that doesn't exist.

I also feel that if you accept these as JSON configs, the content should be customizable by each module, while the designs remain the same across devices which is the opposite of this particular case.

The reason why I suggested a JSON configuration method for the Menu Items was because, the content in it (the menu items) could change from module to module and I wanted to keep one component for the Layout completely and not have it in every single module, so I needed to define it as JSON.
And this was easy to implement as there was a fixed structure to it and could be implemented using the recursive method we discussed a few weeks back.
 
apart from that there, are the changes I requested in  https://github.com/reach4help/reach4help/pull/481#discussion_r419086837